IGI.Lab3/CarPark.WebAPI/Startup.cs

```diff
@@ -0,0 +1,125 @@
+                    .AsSelf()
+                    .WithTransientLifetime())
+                .AddAutoMapper(typeof(MapperProfile), typeof(WebUI.MappingProfiles.MapperProfile))
+                .AddScoped(typeof(IRepository<>), typeof(Repository<>));
+            services.AddControllers();
+            services.AddAuthentication()
+                .AddJwtBearer(configureOptions => configureOptions.TokenValidationParameters =
+                    new TokenValidationParameters
+                    {
+                        ValidIssuer = JwtToken.Issuer,
+                        ValidAudience = JwtToken.Audience,
+                        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(JwtToken.Key))
+                    });
+
+            services.AddSwaggerGen(swaggerOptions =>
+            {
+                swaggerOptions.SwaggerDoc("v1", new OpenApiInfo { Title = "CarParkApi" });
+
+                swaggerOptions.AddSecurityDefinition("Bearer", new OpenApiSecurityScheme
+                {
+                    Description = "JWT Authorization",
+                    Name = "Authorization",
+                    In = ParameterLocation.Header,
+                    Type = SecuritySchemeType.ApiKey,
+                    Scheme = "Bearer"
+
+                });
+
+                swaggerOptions.AddSecurityRequirement(new OpenApiSecurityRequirement
+                {
+                    {
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference
+                            {
+                                Type = ReferenceType.SecurityScheme,
+                                Id = "Bearer"
+                            },
+                            Scheme = "oauth2",
+                            Name =  "Bearer",
+                            In = ParameterLocation.Header
+                        } ,
+                        new List<string>()
+                    }
+                });
+            });
+
+            services.AddIdentity<IdentityUser, IdentityRole>(options =>
+                {
+                    options.Password.RequireNonAlphanumeric = false;
+                })
+                .AddEntityFrameworkStores<CarParkContext>();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseSwagger();
+
+            app.UseSwaggerUI(swaggerOptions =>
+                {
+                    swaggerOptions.SwaggerEndpoint("/swagger/v1/swagger.json", "CarParkApi");
+                    swaggerOptions.RoutePrefix = string.Empty;
+                }
+            );
+
+            app.UseHttpsRedirection();
```

Comment:
Do we need it here?

Reply:
Adds redirection to the same resource for the project only via the https protocol if the application has SSL support. But I do not think that it is necessary in my application now.